src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,149 @@
+
+    // HashMap to hold user IDs and the number of messages sent by that user
+    HashMap<UUID, Integer> userMessageCounts = new HashMap<UUID, Integer>();
+
+    int totalMessageCount = 0; // total number of messages sent by all users
+    int totalWordCount = 0; // total number of words sent by all users
+
+    for (Conversation c: allConversations) {
+
+      messagesInConversation = messageStore.getMessagesInConversation(c.getId());
+      totalMessageCount += messagesInConversation.size();
+
+      for (Message m: messagesInConversation) {
+
+        int tmp = userMessageCounts.getOrDefault(m.getAuthorId(), 0);
+        userMessageCounts.put(m.getAuthorId(), tmp + 1);
+
+        totalWordCount += m.getContent().split("\\s+").length;
+      }
+
     }
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response)
-        throws IOException, ServletException {
+    // compute most active user
+    Map.Entry<UUID, Integer> maxMessageCountUser = null;
 
-        String user = (String) request.getSession().getAttribute("user");
+    for (Map.Entry<UUID, Integer> entry: userMessageCounts.entrySet()) {
 
-        if (admins.contains(user)) {
-            request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
-        } else {
-            response.sendError(HttpServletResponse.SC_NOT_FOUND);
-        }
+      if (maxMessageCountUser == null || entry.getValue() > maxMessageCountUser.getValue()) {
+        maxMessageCountUser = entry;
+      }
 
     }
+    if (maxMessageCountUser != null && userStore.getUser(maxMessageCountUser.getKey()) != null) {
+      mostActiveUser = userStore.getUser(maxMessageCountUser.getKey()).getName();
+    }
+
+    // Set attributes
+    request.setAttribute("numberOfUsers", allUsers.size());
+    request.setAttribute("newestUser", newestUser);
+    request.setAttribute("mostActiveUser", mostActiveUser);
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response)
-        throws IOException, ServletException {
+    request.setAttribute("numberOfConversations", allConversations.size());
+    request.setAttribute("numberOfMessages", totalMessageCount);
+    request.setAttribute("avgMessagesPerConvo", (int) ((double) totalMessageCount / allConversations.size()));
+    request.setAttribute("avgWordsPerMessage", (int) ((double) totalWordCount / totalMessageCount));
 
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+
+    String user = (String) request.getSession().getAttribute("user");
+
+    if (admins.contains(user)) {
+      computeAdminStats(request);
+      request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+    } else {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
     }
+
+  }
+
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
If this servlet doesn't have a POST implementation yet, then no need to add this method.